packages/storefront-sdk-plugins/commerce-queries/src/index.ts

```diff
@@ -203,6 +219,73 @@ function commerceQueriesPlugin<TBase extends WithStorefrontQuery & WithConfig>(
 				)
 			} as StorefrontResponse<ProductEdge[] | Product[]>;
 		}
+
+		async productCollections(params?: FetchProductCollectionsMethodParams) {
+			const {
+				cursor,
+				nacelleEntryIds,
+				handles,
+				locale = this.getConfig()?.locale,
+				maxReturnedEntries = this.#defaultMaxReturnedEntries,
+				maxReturnedEntriesPerCollection,
+				advancedOptions,
+				edgesToNodes = true
+			} = params ?? {};
+
+			if (Number(advancedOptions?.entriesPerPage) > 100) {
+				console.warn(highEntriesPerPageMessage);
+			}
+
+			const first = Math.min(
+				...[
+					advancedOptions?.entriesPerPage ?? this.#defaultPageFetchLimit,
+					maxReturnedEntries
+				].filter((n) => n > 0)
+			);
+
+			const filter: ProductCollectionFilterInput = {
+				after: cursor,
+				locale,
+				first
+			};
+
+			// keeping with v1 sdk, only use nacelleEntryIds if both handles and nacelleEntryIds are provided
+			if (nacelleEntryIds && handles) {
+				console.warn(entryIdsAndHandlesMessage('productCollections'));
+			}
+
+			if (nacelleEntryIds) {
+				filter.nacelleEntryIds = nacelleEntryIds;
+			} else {
+				filter.handles = handles;
+			}
+
+			const responseData = await requestPaginatedData<
+				this,
+				ProductCollection,
+				ProductCollectionEdge,
+				ProductCollectionFilterInput
+			>(
+				this,
+				'allProductCollections',
+				filter,
+				maxReturnedEntries,
+				edgesToNodes,
+				maxReturnedEntriesPerCollection
+			);
+
+			if (responseData?.error) {
+				return responseData;
+			}
+
+			return {
+				data: await (this as unknown as StorefrontClient)['applyAfter'](
+					'productCollections',
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					responseData.data!
+				)
+			} as StorefrontResponse<ContentEdge[] | Content[]>;
```

Comment:
Is this the right type for productCollections?

Reply:
Not in the slightest! Nice catch. Fixed in https://github.com/getnacelle/nacelle-js/pull/323/commits/f86d3693ba12f7b8b59cbe1ac4979aa475ffa20b. Also added some new tests to check that all methods are returning the expected type.